pkg/monitor/monitor_test.go

```diff
@@ -433,8 +516,75 @@ func testOfflineStorageNode(t *testing.T) {
 	require.NoError(t, err, "expected no error from get pod as pod should not be deleted")
 }
 
+func testOfflineStorageNodesBatchTest(t *testing.T) {
+
+	setupWithNewMockDriverScale(t)
+
+	defer teardownWithNewMockDriver(t)
+	numNodes := 50
+	offlineNodes := 12
+	// The test creates the following pods with 1 pod in each node
+	pods := make([]*v1.Pod, 0)
+	for i := 0; i < numNodes; i++ {
+		pod := newPod(fmt.Sprintf("driverPod%d", i), []string{fmt.Sprintf("%s%d", driverVolumeName, i)}, fmt.Sprintf("node%d.domain", i))
+		_, err := core.Instance().CreatePod(pod)
+		require.NoError(t, err, "failed to create pod %s for node %d", pod.Name, i)
+		pods = append(pods, pod)
+	}
+
+	for i := 0; i < numNodes; i++ {
+		podList := v1.PodList{}
+		podList.Items = append(podList.Items, *pods[i])
+		mockCacheInstance.EXPECT().ListTransformedPods(fmt.Sprintf("node%d.domain", i)).Return(&podList, nil).AnyTimes()
+		//mockCacheInstance.EXPECT().GetPersistentVolumeClaim("cvolume1", defaultNamespace).Return(pvc1, nil).Times(1)
+	}
+
+	driverNodes := getDriverNodes(len(nodes.Items))
+	// Mark offlineNodes number of nodes as offline
+	for i := 0; i <= offlineNodes; i++ {
+		driverNodes[i].Status = volume.NodeOffline
+	}
+	volumeDriver.EXPECT().GetNodes().Return(driverNodes, nil).AnyTimes()
+	volumeDriver.EXPECT().GetCSIPodPrefix().Return("px-csi-ext-", nil).AnyTimes()
+
+	for i := 0; i < numNodes; i++ {
+		volumes := []*volume.Info{
+			{
+				VolumeName: fmt.Sprintf("volume%d", i),
+				DataNodes:  []string{driverNodes[0].StorageID},
+			},
+		}
+		wffcVolumes := make([]*volume.Info, 0)
+		volumeDriver.EXPECT().InspectNode(fmt.Sprintf("node%d", i)).Return(driverNodes[i], nil).AnyTimes()
+		volumeDriver.EXPECT().GetPodVolumes(&pods[i].Spec, pods[i].Namespace, false).Return(volumes, wffcVolumes, nil).AnyTimes()
+	}
+
+	// Since the monitor runs every 30 seconds and to ensure with certainilt one batch of pod deletes, waiting for a mx of 95 seconds
```

Comment:
nit.."certainty"  
"max" 
remove "Since"

Reply:
Removed the comment itself and put the break up of wait times